IHP/IDE/CodeGen/ActionGenerator.hs

```diff
@@ -0,0 +1,131 @@
+
+            showContent =
+                ""
+                <> "    action Show" <> singularName <> "Action { " <> idFieldName <> " } = do\n"
+                <> "        " <> modelVariableSingular <> " <- fetch " <> idFieldName <> "\n"
+                <> "        render ShowView { .. }\n"
+
+            editContent =
+                ""
+                <> "    action Edit" <> singularName <> "Action { " <> idFieldName <> " } = do\n"
+                <> "        " <> modelVariableSingular <> " <- fetch " <> idFieldName <> "\n"
+                <> "        render EditView { .. }\n"
+
+            updateContent =
+                ""
+                <> "    action Update" <> singularName <> "Action { " <> idFieldName <> " } = do\n"
+                <> "        " <> modelVariableSingular <> " <- fetch " <> idFieldName <> "\n"
+                <> "        " <> modelVariableSingular <> "\n" 
+                <> "            |> build" <> singularName <> "\n"
+                <> "            |> ifValid \\case\n"
+                <> "                Left " <> modelVariableSingular <> " -> render EditView { .. }\n"
+                <> "                Right " <> modelVariableSingular <> " -> do\n"
+                <> "                    " <> modelVariableSingular <> " <- " <> modelVariableSingular <> " |> updateRecord\n"
+                <> "                    setSuccessMessage \"" <> model <> " updated\"\n"
+                <> "                    redirectTo Edit" <> singularName <> "Action { .. }\n"
+
+            createContent =
+                ""
+                <> "    action Create" <> singularName <> "Action = do\n"
+                <> "        let " <> modelVariableSingular <> " = newRecord @"  <> model <> "\n"
+                <> "        " <> modelVariableSingular <> "\n" 
+                <> "            |> build" <> singularName <> "\n"
+                <> "            |> ifValid \\case\n"
+                <> "                Left " <> modelVariableSingular <> " -> render NewView { .. } \n"
+                <> "                Right " <> modelVariableSingular <> " -> do\n"
+                <> "                    " <> modelVariableSingular <> " <- " <> modelVariableSingular <> " |> createRecord\n"
+                <> "                    setSuccessMessage \"" <> model <> " created\"\n"
+                <> "                    redirectTo " <> name <> "Action\n"
+
+            deleteContent =
+                ""
+                <> "    action Delete" <> singularName <> "Action { " <> idFieldName <> " } = do\n"
+                <> "        " <> modelVariableSingular <> " <- fetch " <> idFieldName <> "\n"
+                <> "        deleteRecord " <> modelVariableSingular <> "\n"
+                <> "        setSuccessMessage \"" <> model <> " deleted\"\n"
+                <> "        redirectTo " <> name <> "Action\n"
+
+            typesContent = 
+                   "    | " <> nameWithSuffix
```

Comment:
When I generate a `ShowUser` action, this will generate the data constructor:

```
| ShowUserAction
```

But it should be the following:

```
| ShowUserAction { userId :: !(Id User) }
```